crd/multitenantnetworkcontainer/manifests/networking.azure.com_multitenantnetworkcontainers.yaml

```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.12.0
+    controller-gen.kubebuilder.io/version: v0.16.5
```

Comment:
`controller-gen` et al are versioned to support specific kubernetes versions. the 1.4 branch is for 1.27, and we should not update any k8s deps past the versions which support or were built against 1.27...
that said, this diff looks harmless so updating the tool here is probably fine. but that's why it hadn't been updated previously.

Reply:
oops, yeah i avoided upgrading a bunch of other `sigs.k8s.io` dependencies since I'm aware of the version ties but missed this for some reason 🥲